docs: add starter for plugins with new plugin recipes (#22258)
gatsbyjs/gatsby: docs/docs/recipes.md

- [Using Google Fonts](/docs/recipes/styling-css#using-google-fonts)
- [Using Font Awesome](/docs/recipes/styling-css#using-fontawesome)

> ## [3. Working with starters](/docs/recipes/working-with-plugins)

Author: Haha oh wow really missed the text on this section, I only updated the links 😅 

```suggestion
## [3. Working with plugins](/docs/recipes/working-with-plugins)
```

> [Starters](/docs/starters/) are boilerplate Gatsby sites maintained officially, or by the community.

Author: ```suggestion
[Plugins](/docs/plugins/) are Node.js packages that implement Gatsby APIs that are maintained officially, or by the community.
```

> - [Using a starter](/docs/recipes/working-with-plugins#using-a-plugin)

Author: ```suggestion
- [Using a plugin](/docs/recipes/working-with-plugins#using-a-plugin)
```
